src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -157,6 +161,157 @@ namespace OpenMS
       }
     }
   }
+  
+  void MRMTransitionGroupPicker::calculatePeakApexInt_(const MSChromatogram& chromatogram,
+    double best_left, double best_right, 
+    ConvexHull2D::PointArrayType & hull_points,
+    double & intensity_sum, 
+    double & rt_sum,
+    double & peak_apex_int,
+    double & peak_apex_rt)
+  {  
+    intensity_sum = 0.0;
+    rt_sum = 0.0;
+    peak_apex_int = -1;
+    double peak_apex_dist = std::fabs(chromatogram.begin()->getMZ() - peak_apex_rt);
+    // FEATURE : use RTBegin / MZBegin -> for this we need to know whether the template param is a real chromatogram or a spectrum!
+    for (MSChromatogram::const_iterator it = chromatogram.begin(); it != chromatogram.end(); ++it)
+    {
+      if (it->getMZ() > best_left && it->getMZ() < best_right)
+      {
+        DPosition<2> p;
+        p[0] = it->getMZ();
+        p[1] = it->getIntensity();
+        hull_points.push_back(p);
+        if (std::fabs(it->getMZ() - peak_apex_rt) <= peak_apex_dist)
+        {
+          peak_apex_int = p[1];
+          peak_apex_dist = std::fabs(it->getMZ() - peak_apex_rt);
+        }
+        rt_sum += it->getMZ();
+        intensity_sum += it->getIntensity();
```

Comment:
@dmccloskey just to highlight again, this is how I "integrate" the peak: there is no integration, its a simple sum. Note that therefore this only works if we use the same sampling speed across multiple runs, otherwise we run into problems here. The correct way to do this would be to indeed perform a trapezoidal peak integration here.

Reply:
I see.  Ok this would explain the discrepency that I was finding between the areas reported.  I was using peak height, so it was not a big deal, but using peak area is much more common than peak height.

Reply:
I think also for peak height, a model that interpolates between the two most intense peaks may be beneficial, e.g. a spline interpolation -- as the "actual" peak height is usually higher than the one observed due to undersampling

Reply:
This is very true.  What might be even better that you discussed previously was fitting the data points to a Gaussian or some other model.  I think this would be a great approach.  I have a lot of manually integrated peaks (~100,000) generated from standards as well as an assortment of biological matrices that could be used for fitting such a model as well as validation.

Reply:
sounds like a great resource to test and validate these models, so these standards have known concentration ratios?